linkerd/app/inbound/src/metrics/authz.rs

```diff
@@ -35,48 +37,62 @@ pub(crate) struct TcpAuthzMetrics(Arc<TcpInner>);
 
 #[derive(Debug, Default)]
 struct HttpInner {
-    allow: Mutex<HashMap<AuthzKey, Counter>>,
-    deny: Mutex<HashMap<SrvKey, Counter>>,
+    allow: Mutex<HashMap<RouteAuthzKey, Counter>>,
+    deny: Mutex<HashMap<RouteKey, Counter>>,
 }
 
 #[derive(Debug, Default)]
 struct TcpInner {
-    allow: Mutex<HashMap<AuthzKey, Counter>>,
+    allow: Mutex<HashMap<ServerAuthzKey, Counter>>,
     deny: Mutex<HashMap<SrvKey, Counter>>,
     terminate: Mutex<HashMap<SrvKey, Counter>>,
 }
 
 #[derive(Debug, Hash, PartialEq, Eq)]
 struct SrvKey {
     target: TargetAddr,
-    server: ServerLabel,
+    labels: ServerLabel,
     tls: tls::ConditionalServerTls,
 }
 
 #[derive(Debug, Hash, PartialEq, Eq)]
-struct AuthzKey {
+struct ServerAuthzKey {
     target: TargetAddr,
     authz: ServerAuthzLabels,
     tls: tls::ConditionalServerTls,
 }
 
+#[derive(Debug, Hash, PartialEq, Eq)]
+struct RouteKey {
+    target: TargetAddr,
+    tls: tls::ConditionalServerTls,
+    labels: RouteLabels,
+}
+
+#[derive(Debug, Hash, PartialEq, Eq)]
+struct RouteAuthzKey {
+    target: TargetAddr,
+    tls: tls::ConditionalServerTls,
+    labels: RouteAuthzLabels,
+}
```

Comment:
kinda wondered if it would make sense to do something like
```rust
#[derive(Debug, Hash, PartialEq, Eq)]
struct Key<T> {
    target: TargetAddr,
    tls: tls::ConditionalServerTls,
    labels: T,
}

type ServerAuthzKey = Key<ServerAuthzLabels>;
type RouteKey = Key<RouteLabels>;
type RouteAuthzKey = Key<RouteAuthzLabels>;
```
but, not sure if it's worth it.

Reply:
Yeah this cleans up a lot actually